### DISCUSSION TITLE
c_cpp_properties.json Dosyası ile İlgili

### DISCUSSION THREAD

**Discussion by furkan-yalcin** [19/classroom#31] [Anonim] [answered]
Ödevi Classroom ortamında yaptım. `.vscode` dizininin içerisine `c_cpp_properties.json` adında bir dosya eklendi. Dosyanın içeriği görseldeki gibidir.
![dis1](https://user-images.githubusercontent.com/91963128/158624791-6bdc0688-0384-41bd-b871-24e47514b6e8.png)

Bu dosyanın nereden geldiğini araştırdığımda şöyle bir kaynakla karşılaştım : [https://code.visualstudio.com/docs/cpp/config-msvc#:~:text=C%2FC%2B%2B%20configurations%23,%2B%2B17)%2C%20and%20more.](url)

Kaynak, C/C++ Configurations sayfasında değişiklik yapıldığında VS Code' un `.vscode` dizininin içerisine `c_cpp_properties.json` adında bir dosya yazacağını söylüyor.
Ben C/C++ Configurations sayfasında herhangi bir değişiklik yapmamama rağmen bu dosya eklendi.

Sorularım şunlar :

1. Bu dosya kurmuş olduğum eklentilerden biri yüzünden eklenmiş olabilir mi ? Kurduğum eklentiler görseldeki gibidir.
 ![diss4](https://user-images.githubusercontent.com/91963128/158633483-a19a0468-0b71-4311-93d9-6988feace4f5.png)

2. Bize `.` ile başlayan dosyalarda ve dizinlerde değişiklik yapmamamız söylendi. Ödevi commit' lediğimde depodaki `.vscode` dizininin içerisine `c_cpp_properties.json` dosyası da commit' lenmiş oldu.
![diss2](https://user-images.githubusercontent.com/91963128/158634318-c64daa86-d728-4565-8704-bebc93eb3374.png)
![diss3](https://user-images.githubusercontent.com/91963128/158634327-8e81f89c-14d5-49f5-a939-f962737d28bf.png)

Bu, ödevler notlandırılırken bir sorun teşkil eder mi?

**Comment by roktas** [comment]
Sorun teşkil etmez ama sorayım: [bu dokümanı](https://classroom.alaturka.dev/guide/#vs-code-icinden-wsl-e-erisim) okudun mu?

**Reply by furkan-yalcin** [reply]
Evet hocam okudum.

**Comment by roktas** [comment, marked as answer]
> Ben C/C++ Configurations sayfasında herhangi bir değişiklik yapmamama rağmen bu dosya eklendi.

Bu dosya kendi kendine eklenmez.  Bir yerde onay istemiştir senden, sen de farkında olmadan onay vermişsindir.

İlke açık: sadece sizden istenen dosyalar içinde çalışın.  Bu ilk ödev olduğundan dikkate almayacağım ama gelecekte daha dikkatli ol.

**Reply by furkan-yalcin** [reply]
Tamamdır hocam yanıtınız için teşekkür ederim.
